modules/telemetry/telemetry.php

```diff
@@ -0,0 +1,232 @@
+		add_action( 'vw_delete_custom_status', [ __CLASS__, 'record_delete_custom_status' ], 10, 2 );
+		add_action( 'vw_update_custom_status', [ __CLASS__, 'record_update_custom_status' ], 10, 2 );
+
+		// Notification events
+		add_action( 'vw_notification_status_change', [ __CLASS__, 'record_notification_sent' ], 10, 3 );
+
+		// Settings events
+		add_action( 'vw_upgrade_version', [ __CLASS__, 'record_admin_update' ], 10, 2 );
+		add_action( 'vw_save_settings', [ __CLASS__, 'record_settings_update' ], 10, 2 );
+	}
+
+	// Custom Status events
+
+	/**
+	 * Record an event when a post's custom status changes
+	 *
+	 * @param string $new_status The new status
+	 * @param string $old_status The old status
+	 * @param WP_Post $post The post object
+	 */
+	public static function record_custom_status_change( string $new_status, string $old_status, WP_Post $post ): void {
+		if ( ! in_array( $post->post_type, HelperUtilities::get_supported_post_types() ) ) {
+			// This isn't a supported post type
+			return;
+		} elseif ( $old_status === $new_status ) {
+			// The status hasn't changed
+			return;
+		} elseif ( in_array( $new_status, [ 'inherit', 'auto-draft' ] ) ) {
+			// The status hasn't changed, or it moved into an auto-generated status
+			return;
+		}
+
+		$status_slugs = wp_list_pluck( CustomStatus::get_custom_statuses(), 'slug' );
+
+		if ( ! in_array( $new_status, $status_slugs, true ) && ! in_array( $old_status, $status_slugs, true ) ) {
+			// The status isn't moving to or from a custom status
+			return;
+		}
+
+		self::$tracks->record_event( 'post_custom_status_changed', [
+			'old_status' => $old_status,
+			'new_status' => $new_status,
+			'post_id'    => $post->ID,
+		] );
+	}
+
+	/**
+	 * Record an event when a custom status is created
+	 *
+	 * @param int $term_id The term's ID
+	 * @param string $term_name The term's name
+	 * @param string $slug The term's slug
+	 */
+
+	public static function record_add_custom_status( int $term_id, string $term_name, string $term_slug ): void {
+		self::$tracks->record_event( 'custom_status_created', [
+			'term_id' => $term_id,
+			'name'    => $term_name,
+			'slug'    => $term_slug,
+		] );
+	}
+
+	/**
+	 * Record an event when a custom status is deleted
+	 *
+	 * @param int $term_id The custom status term ID
+	 * @param string $slug The status slug
+	 */
+	public static function record_delete_custom_status( int $term_id, string $slug ): void {
+		self::$tracks->record_event( 'custom_status_deleted', [
+			'term_id' => $term_id,
+			'slug'    => $slug,
+		] );
+	}
+
+	/**
+	 * Record an event when a custom status is updated
+	 *
+	 * @param WP_Term $updated_status The updated status WP_Term object.
+	 * @param array $update_args The arguments used to update the status.
+	 */
+	public static function record_update_custom_status( WP_Term $updated_status, array $update_args ): void {
+		$is_position_update = 1 === count( $update_args ) && isset( $update_args['position'] );
```

Comment:
Here we use the passed-in `$update_args` to determine if there's just a position switch happening. Before this code was added, any reorder caused an event to be tracked for every status.